moto/server.py

```diff
@@ -102,6 +102,10 @@ def infer_service_region_host(self, environ):
                 # If Newer API version, use dynamodb2
                 if dynamo_api_version > "20111205":
                     host = "dynamodb2"
+        elif service == "sagemaker":
+            host = "api.sagemaker.{region}.amazonaws.com".format(
```

Comment:
Nice find - that must have been a fun bug to track down

Reply:
Just call it an educational opportunity.  ;)